pkg/bpf/exporter.go

```diff
@@ -66,11 +56,10 @@ type exporter struct {
 	cpuCores              int
 	// due to performance reason we keep an empty struct to verify if a new read is also empty
 	emptyct                 ProcessBPFMetrics
-	hardwareCountersEnabled bool
 	byteOrder               binary.ByteOrder
 	perfEventFds            []int
-	enabledHardwareCounters []string
-	enabledSoftwareCounters []string
+	enabledHardwareCounters map[string]struct{}
+	enabledSoftwareCounters map[string]struct{}
```

Comment:
since we already depend on apimachinery, https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set fits perfectly here.